### PR TITLE
Make CI more manageable 

### LINKED ISSUE #2719: migrate docker tests
### What is the problem
The old separate docker repo contains test files, to ensure that the images we publish are working as intended. We need to migrate these over, and integrate them into the new maven way of things.

The question is how to integrated them either as shell tests executed by a shell plugin, or by rewriting them as java tests. 

The second question is how these tests should integrate with other test to be run as integration tests via maven

### What did you expect
Published images to be tested.

### Context information
Please always add the following information
- eXist-db version 5.0.0-RC8-SNAPSHOT

### LINKED ISSUE #3696: [feature] re-enable docker builds
Since the changes to `travis-ci.com` on `01.11.2020` we no longer publish docker images. So wherever exist goes from here, we need to reactivate the docker publishing pipeline.

### PR DESCRIPTION
**Pull request by duncdrum**
Speed up ci by increasing parallelism, giving us more control over restarting Workflows

add unit test for docker images and run them on ci

close #2719
close #3696

see #1771
see #2987

### CODE REVIEW

File: exist-docker/src/test/bats/02-config-spec.bats
Comment (line-o) [inline exist-docker/src/test/bats/02-config-spec.bats:45]: Typo "system"

**Review by line-o — COMMENTED** [review]
That looks promising. A lot of good stuff. Can't wait to see the final version.

**Comment by duncdrum** [comment]
I've tested publishing from this script to my own repo and it works, so the message about lacking credentials means the wrong credentials are in this repo. I cannot edit things on docker hub. So adding mine here won't do the trick. 

To make sure we have the right credentials, simply restart the `deploy` workflow after editing them. 

Once this is merged we should swiftly comment out [L55-66](]https://github.com/eXist-db/exist/actions/runs/630622367/workflow#L59-L66) there is no harm in pushing `experimental` tags it simply adds noise after the first one or two.

**Comment by dizzzz** [comment]
image

**Comment by duncdrum** [comment]
It just occurred to me that secrets being unavailable from PRs might be a feature of gha secrets. Since we're not deploying from develop right now, I vote to merge this and see if that fixes our problem.

**Comment by adamretter** [comment]
@duncdrum I don't think that can be the case, as for the SonarCloud one we had a secret key we had to set `SONAR_TOKEN` and as far as I can see, that worked fine in a PR.

**Comment by duncdrum** [comment]
@adamretter the sonar token is an org secret, the docker stuff is a repo secret. GH-A handles secrets differently compared to Travis, Circle etc so i m still trying to figure it out. But I strongly suspect that to be the case. I would like to see a run on develop first to confirm, so we can adjust accordingly later

**Comment by adamretter** [comment]
Okay can you prepare a pr which is s single commit and just the deploy step, I can merge that to develop for testing. This pr mixes in other stuff and I'd like to keep it lean for testing on develop

**Comment by dizzzz** [comment]
image

**Review by dizzzz — CHANGES_REQUESTED** [review]
need rework

**Comment by duncdrum** [comment]
rebase and its running, once this is merged we should consider removing appveyor either all together, or at least limits its matrix, as it is slowing us down and just duplicating environments.

**Comment by adamretter** [comment]
@duncdrum I reworded your commit messages to match the documented standard for eXist-db. Assuming that CI passes again, I will get this merged...

**Pull request merged**
